### ISSUE TITLE
When building image whole tar stores in memory

### ISSUE THREAD

**Issue by hh-h** [aio-libs/aiodocker#352]
Hi, we create huge docker images from tar files, up to 10 GB, it's kind of problematic because now it stores whole tar file in memory, see https://github.com/aio-libs/aiodocker/blob/11fa31c807c438281569a61fcf3ea4e24deb5258/aiodocker/images.py#L232-L234
I tried to experiment and changed it to simple "async" generator, since aiohttp can accept that as a data parameter, like this
```python
async def _stream(self, f):
    import io
    chunk = f.read(io.DEFAULT_BUFFER_SIZE)
    while chunk:
        yield chunk
        chunk = f.read(io.DEFAULT_BUFFER_SIZE)
```
and replaced
`local_context = fileobj.read()`
with
`local_context = self._stream(fileobj)`
at least tests are passed

I think it could block somewhere in async code somehow, but I didn't dig into.

What do you think?

**Comment by asvetlov** [comment]
The fix looks correct. Please prepare a PR.

I'm curious is it 10GB tarfile or 10GB image with a huge base layer?
I encountered only the last case.

**Comment by hh-h** [comment]
It's 10GB tarfile  
The idea is that we have a base image, then users can add files to some folder, we put dockerfile to that folder and extend base image with these user files.

**Comment by asvetlov** [comment]
The size is a little crazy. Is it for data science?

**Comment by hh-h** [comment]
We're building a testing framework around JMeter and Selenium. In this tar file users store files that needed for tests, that's all that I can say :)

**Comment by asvetlov** [comment]
Got you, thanks.
It really doesn't matter why do you need such huge image.
I just was curious.

**Comment by hh-h** [comment]
Closed #353
